src/material-experimental/mdc-dialog/dialog-content-directives.ts

```diff
@@ -0,0 +1,145 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import {
+  Directive,
+  ElementRef,
+  Input,
+  OnChanges,
+  OnInit,
+  Optional,
+  SimpleChanges,
+} from '@angular/core';
+
+import {MatDialog} from './dialog';
+import {MatDialogRef} from './dialog-ref';
+
```

Comment:
It looks like these directives are the same as the ones in `dialog-content-directives.ts`. Couldn't we pull those directives out into a module that is imported here and in the base dialog? This would help us avoid the duplication.

Reply:
I intentionally copied these small directives over. Mostly due to the overhead of splitting the directives into a base class without metadata. I can surely split them if you think it's necessary.

Reply:
I meant just reusing the same directives, not extending them. From what I can tell the behavior is exactly the same. Here's an example of how we're reusing some of the menu directives: https://github.com/angular/components/blob/master/src/material/menu/menu-module.ts#L31

Reply:
The directives are different though. We have different host bindings, default ids, different DI tokens for the dialog, and logic for determining the associated dialog.